shared/pingpong/accounts.go

```diff
@@ -186,21 +194,36 @@ func prepareAssets(accounts map[string]uint64, client libgoal.Client, cfg PpConf
 			totalSent++
 			throttleTransactionRate(startTime, cfg, totalSent)
 		}
-		account, accountErr := client.AccountInformation(addr)
-		if accountErr != nil {
-			fmt.Printf("Cannot lookup source account %v\n", addr)
-			err = accountErr
-			return
-		}
+	}
 
+	// wait until all the assets created
+	for addr := range accounts {
+		var account v1.Account
+		for {
+			account, err = client.AccountInformation(addr)
+			if err != nil {
+				fmt.Printf("Warning, cannot lookup source account")
+				return
+			}
+			if len(account.AssetParams) >= assetsInAcct[addr] {
```

Comment:
The idea is we know that we should create `assetsInAcct[addr]` assets and we wait until `len(account.AssetParams)` has all of them - it can have less if creation transactions are pending.
But. you are right, `assetsInAcct[addr]` needs to be initialized to `int(cfg.NumAsset) + len(addrAccount.AssetParams)` sinse some assets might pre-exist.